src/ctap/mod.rs

```diff
@@ -344,27 +344,25 @@ fn check_user_presence(env: &mut impl Env, channel: Channel) -> Result<(), Ctap2
                 KEEPALIVE_DELAY
             },
         );
-        if let Ok(UserPresenceStatus::Timeout) = result {
-        } else {
+        if !matches!(result, Err(UserPresenceError::Timeout)) {
             break;
         }
         // TODO: this may take arbitrary time. Next wait's delay should be adjusted
         // accordingly, so that all wait_with_timeout invocations are separated by
         // equal time intervals. That way token indicators, such as LEDs, will blink
         // with a consistent pattern.
         if send_keepalive_up_needed(env, channel, KEEPALIVE_DELAY).is_err() {
-            result = Ok(UserPresenceStatus::Canceled);
+            result = Err(UserPresenceError::Canceled);
             break;
         }
     }
 
     env.user_presence().check_complete(&result);
     match result {
-        Ok(UserPresenceStatus::Timeout) => Err(Ctap2StatusCode::CTAP2_ERR_USER_ACTION_TIMEOUT),
-        Ok(UserPresenceStatus::Declined) => Err(Ctap2StatusCode::CTAP2_ERR_OPERATION_DENIED),
-        Ok(UserPresenceStatus::Confirmed) => Ok(()),
-        Ok(UserPresenceStatus::Canceled) => Err(Ctap2StatusCode::CTAP2_ERR_KEEPALIVE_CANCEL),
-        Err(_) => panic!("Unhandled error checking for user presence"),
+        Ok(()) => Ok(()),
+        Err(UserPresenceError::Timeout) => Err(Ctap2StatusCode::CTAP2_ERR_USER_ACTION_TIMEOUT),
+        Err(UserPresenceError::Declined) => Err(Ctap2StatusCode::CTAP2_ERR_OPERATION_DENIED),
+        Err(UserPresenceError::Canceled) => Err(Ctap2StatusCode::CTAP2_ERR_KEEPALIVE_CANCEL),
```

Comment:
Optional: Could move into a `From` implementation in `status_code.rs`.

Reply:
Done